src/Microsoft.AspNet.Razor.Runtime/Resources.resx

```diff
@@ -140,4 +140,7 @@
   <data name="TagHelperDescriptorResolver_EncounteredUnexpectedError" xml:space="preserve">
     <value>Encountered an unexpected error when attempting to resolve tag helper directive '{0}' with value '{1}'. Error: {2}</value>
   </data>
+  <data name="HtmlElementNameAttribute_InvalidElementName" xml:space="preserve">
+    <value>Tag helpers cannot target element name "{0}".</value>
```

Comment:
- use single quotes to quote values
- you're passing `"!"` rather than the invalid element name
- give the user more information e.g. ... `element name '{0}' because it contains a '{1}' character.`